inc/inventory/asset/mainasset.class.php

```diff
@@ -444,8 +447,12 @@ public function handle() {
             //no rule matched, this is a new one
             $this->rulepassed(0, $this->item->getType(), null);
          } else if (!isset($datarules['found_inventories'])) {
-             $input['rules_id'] = $datarules['rules_id'];
-             $this->addRefused($input);
+            if (!property_exists($data, 'is_ap') || $data->is_ap != true) {
+               $input['rules_id'] = $datarules['rules_id'];
+               $this->addRefused($input);
+            } else {
+               unset($this->data[$key]);
+            }
```

Comment:
To be sure that everyone will understand this piece of code in a far future, I would change it to:
```suggestion
            if (property_exists($data, 'is_ap') || $data->is_ap) {
               // A comment to explain why AP are handled differently.
               unset($this->data[$key]);
            } else {
               $input['rules_id'] = $datarules['rules_id'];
               $this->addRefused($input);
            }
```

Reply:
done